xapian-glib/xapian-term-iterator.cc

```diff
@@ -0,0 +1,317 @@
+
+G_DEFINE_TYPE_WITH_PRIVATE (XapianTermIterator, xapian_term_iterator, G_TYPE_OBJECT)
+
+static void
+xapian_term_iterator_finalize (GObject *gobject)
+{
+  XapianTermIteratorPrivate *priv = XAPIAN_TERM_ITERATOR_GET_PRIVATE (gobject);
+
+  delete priv->data;
+
+  G_OBJECT_CLASS (xapian_term_iterator_parent_class)->finalize (gobject);
+}
+
+static void
+xapian_term_iterator_class_init (XapianTermIteratorClass *klass)
+{
+  G_OBJECT_CLASS (klass)->finalize = xapian_term_iterator_finalize;
+}
+
+static void
+xapian_term_iterator_init (XapianTermIterator *self)
+{
+}
+
+/*< private >
+ * xapian_term_iterator_new:
+ * @it: a Xapian::TermIterator
+ *
+ * Creates a new #XapianTermIterator from the given `Xapian::TermIterator` instance.
+ *
+ * Returns: (transfer full): the newly created #XapianTermIterator
+ *
+ * Since: 1.4
+ */
+XapianTermIterator *
+xapian_term_iterator_new (const Xapian::TermIterator &it)
+{
+  XapianTermIterator *iter;
+  XapianTermIteratorPrivate *priv;
+
+  iter = static_cast<XapianTermIterator *> (g_object_new (XAPIAN_TYPE_TERM_ITERATOR, NULL));
+  priv = XAPIAN_TERM_ITERATOR_GET_PRIVATE (iter);
+  priv->data = new IteratorData (it);
+
+  return iter;
+}
+
+/**
+ * xapian_term_iterator_is_valid:
+ * @iter: a #XapianTermIterator
+ *
+ * Checks whether @iter is valid.
+ *
+ * Returns: %TRUE if the iterator is valid
+ *
+ * Since: 1.4
+ */
+gboolean
+xapian_term_iterator_is_valid (XapianTermIterator *iter)
+{
+  g_return_val_if_fail (XAPIAN_IS_TERM_ITERATOR (iter), FALSE);
+
+  return XAPIAN_TERM_ITERATOR_GET_PRIVATE (iter)->data != NULL;
+}
+
+/**
+ * xapian_term_iterator_next:
+ * @iter: a #XapianTermIterator
+ *
+ * Advances the @iter forward.
+ *
+ * You should use the return value to check if @iter is still
+ * valid after being advanced, before using the #XapianTermIterator
+ * API.
+ *
+ * Returns: %TRUE if the iterator was advanced, and %FALSE otherwise
+ *
+ * Since: 1.4
+ */
+gboolean
+xapian_term_iterator_next (XapianTermIterator *iter)
+{
+  g_return_val_if_fail (XAPIAN_IS_TERM_ITERATOR (iter), FALSE);
+
+  XapianTermIteratorPrivate *priv = XAPIAN_TERM_ITERATOR_GET_PRIVATE (iter);
+
+  if (priv->data == NULL)
+    return FALSE;
+
+  return priv->data->next ();
+}
+
+/**
+ * xapian_term_iterator_get_term_name:
+ * @iter: a #XapianTermIterator
+ *
+ * Retrieves the term name of the current item pointed by @iter.
+ *
+ * Returns: (transfer full): the term name
+ *
+ * Since: 1.4
+ */
+char *
+xapian_term_iterator_get_term_name (XapianTermIterator *iter)
+{
+  g_return_val_if_fail (XAPIAN_IS_TERM_ITERATOR (iter), 0);
+
+  XapianTermIteratorPrivate *priv = XAPIAN_TERM_ITERATOR_GET_PRIVATE (iter);
+
+  if (priv->data == NULL)
+    return NULL;
+
+  return priv->data->getTermName ();
+}
+
+/**
+ * xapian_term_iterator_get_wdf:
+ * @iter: a #XapianTermIterator
+ * @error: return location for a #GError
+ *
+ * Retrieves the WDF of the current item pointed by @iter.
```

Comment:
I agree with @cosimoc; it's fine to leave it as is. If you're using xapian-glib we assume you understand what xapian does. :-)